### PR TITLE
feature/add-github-actions

### PR DESCRIPTION
**Pull request by DmitryOnopriienko**
@VolodymyrM27

### CODE REVIEW

File: .github/workflows/detekt.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: oracle
+          java-version: '17'
+
+      - name: Run detekt
+        run: ./gradlew detekt
```
Comment (VolodymyrM27) [inline .github/workflows/detekt.yaml]: `./gradlew build`

better, u can compile + run all checks

File: .github/workflows/detekt.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Detekt-Actions
```
Comment (VolodymyrM27) [inline .github/workflows/detekt.yaml]: you can give a more general name like `Checks` or something like that

Comment (DmitryOnopriienko) [reply .github/workflows/detekt.yaml]: I can, but I like concrete naming

Comment (VolodymyrM27) [reply .github/workflows/detekt.yaml]: Now that you've changed from `./gradlew detekt` to `./gradlew build`, you have a lot of processes going on besides running the detekt plugin to check, at least running tests (if they exist), so a more generic name would be better in this case in my opinion

File: .github/workflows/detekt.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: oracle
+          java-version: '17'
+
+      - name: Run detekt
+        run: ./gradlew build
```
Comment (ajax-gnatiuk-s) [inline .github/workflows/detekt.yaml]: `./gradlew build` does much more than just `detekt`. 

The name of this step doesn't completely reflect what it really does

File: .github/workflows/detekt.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: Detekt-Actions
+
+on:
+  push:
+    branches:
+      - main
+      - 'feature/**'
```
Comment (ajax-gnatiuk-s) [inline .github/workflows/detekt.yaml]: I expect that the job is triggered on:
- **push** to main
- **pull request** to any branch

Comment (DmitryOnopriienko) [reply .github/workflows/detekt.yaml]: Changed trigger to push and PR to any branch

File: .github/workflows/detekt.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: oracle
+          java-version: '17'
+
+      - name: Run detekt
```
Comment (VolodymyrM27) [inline .github/workflows/detekt.yaml]: In this step, you run the build of the entire project, not just the detekt plugin

**Review by VolodymyrM27 — APPROVED** [review]

**Review by ajax-gnatiuk-s — APPROVED** [review]

**Pull request merged**